src/InnerList.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace gapple\StructuredFields;
+
+class InnerList implements TupleInterface, \IteratorAggregate
+{
+    use TupleTrait;
+
+    public function __construct(array $value, ?object $parameters = null)
+    {
+        $this->value = $value;
```

Comment:
This should validate values like `OuterList` does, but not allow nested `InnerList`